crates/hir/src/semantics.rs

```diff
@@ -103,6 +103,26 @@ impl PathResolution {
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct PathResolutionPerNs {
+    pub type_ns: Option<PathResolution>,
+    pub value_ns: Option<PathResolution>,
+    pub macro_ns: Option<PathResolution>,
+}
+
+impl PathResolutionPerNs {
+    pub fn new(
+        type_ns: Option<PathResolution>,
+        value_ns: Option<PathResolution>,
+        macro_ns: Option<PathResolution>,
+    ) -> Self {
+        PathResolutionPerNs { type_ns, value_ns, macro_ns }
+    }
+    pub fn take_path(&self) -> Option<PathResolution> {
```

Comment:
```suggestion
    pub fn any(&self) -> Option<PathResolution> {
```

`take` methods usually modify the object (e.g. `Option::take()`), I propose to call this `any()`.

Reply:
commit : https://github.com/rust-lang/rust-analyzer/pull/19793/commits/bf13549ab639d142342255109725351dbba1288c
rename from `take_path` to `any`